### DISCUSSION TITLE
Support JSON5 for biome.json

### DISCUSSION THREAD

**Discussion by quantizor** [biomejs/biome#282] [Ideas]
It would be great to be able to put code comments in biome.json to label rules with reasons for settings.

**Comment by ematipico** [comment]
We decided to refrain from doing so because that would put pressure on other developer tools (mainly editors), to support them too. Especially for comments in a `.json` file, which isn't valid.

**Reply by quantizor** [reply]
How about a different file format that supports comments then?

**Reply by ematipico** [reply]
I think someone proposed `biome.jsonc` in the Discord server. 

I wouldn't mind, but this comes with a price:
- what's the priority between these file extensions? In case we have both of them in the same directory;
- can `biome.jsonc` extend a `biome.json` file or vice-versa? 

And I would like to have things clarified first.

**Reply by GabenGar** [reply]
1. No priority, the program should crash if it sees two configs with the same base name but different supported extensions in the same folder and tell the users about it.
2. If they have different base names, why not? The parent config has to be parsed first anyway as per the rule above.

**Reply by osdiab** [reply]
I consider this relatively important since, using ESLint at my company, if we switched then we'd lose the institutional knowledge for why we configured it that way, disabled certain rules, etc. @GabenGar 's suggestions make total sense to me.

**Comment by ematipico** [comment]
Closing, track progress in https://github.com/biomejs/biome/issues/1259

**Comment by motlin** [comment]
jsonc and json5 are two different formats, where jsonc syntax is a valid subset of json5 syntax. So I'd like to propose re-opening this issue. The file extension `*.json5` would be required.

**Reply by ematipico** [reply]
Open a new discussion. What OP suggested was/is different from what you suggest
